web/ui/react-app/src/pages/targets/EndpointLink.tsx

```diff
@@ -0,0 +1,31 @@
+import * as React from 'react';
+import SeriesLabel from './SeriesLabel';
+import ErrorAlert from '../../ErrorAlert';
+
+export interface EndpointLinkProps {
+  endpoint: string;
+}
+
+const EndpointLink: React.FC<EndpointLinkProps> = ({ endpoint }) => {
+  let url: URL;
+  try {
+    url = new URL(endpoint);
+  } catch (e) {
+    return <ErrorAlert summary={e.message} />;
+  }
+
+  const { host, pathname, protocol, searchParams }: URL = url;
+  const params = Array.from(searchParams.entries());
+
+  return (
+    <>
+      <a href={endpoint}>{`${protocol}//${host}${pathname}`}</a>
+      {params.length > 0 ? <br /> : null}
+      {params.map(([labelKey, labelValue]: [string, string]) => {
```

Comment:
If you're going for `labelKey` / `labelValue`, then the Prometheus terminology for the key would be label name, thus `labelName`.

Reply:
will do